lib/chewy/index/params.rb

```diff
@@ -0,0 +1,47 @@
+module Chewy
+  class Index
+    class Params
+      STORAGE_INDEX_NAME = 'chewy_index_params'.freeze
+      STORAGE_TYPE_NAME = 'index_params'.freeze
+
+      def self.storage_index
+        @storage_index ||= Class.new(Chewy::Index) do
+          index_name STORAGE_INDEX_NAME
+          define_type STORAGE_TYPE_NAME do
+            field :value, type: 'string', index: 'not_analyzed'
+          end
+        end
+      end
+
+      def self.storage
+        storage_index.type_hash[STORAGE_TYPE_NAME]
+      end
+
+      delegate :storage, to: 'self.class'
+
+      def initialize(index)
+        @index = index
+      end
+
+      def store
+        storage.import([
+          id: @index.derivable_index_name,
+          value: current.to_json
+        ], journal: false)
+      end
+
+      def stored
+        filter = {ids: {values: [@index.derivable_index_name]}}
+        JSON.parse(storage.filter(filter).first.try!(:value) || '{}')
+      end
+
+      def current
+        @index.params_hash.as_json
+      end
+
+      def changed?
+        current != stored
```

Comment:
Yeah, this is covered in specs, so we will notice if the behavior will change :)